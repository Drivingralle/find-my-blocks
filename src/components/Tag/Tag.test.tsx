import React from "react";
import renderer from "react-test-renderer";
import { Tag } from "./";

it("render correctly", () => {
  const tree = renderer.create(<Tag variation="default">Tag</Tag>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("render with an icon", () => {
  const tree = renderer
    .create(
      <Tag variation="default" icon="play">
        Tag
      </Tag>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("render different variations", () => {
  const tag = renderer.create(<Tag variation="default">Tag</Tag>).toJSON();
  const info = renderer.create(<Tag variation="info">Tag</Tag>).toJSON();
  const warning = renderer.create(<Tag variation="warning">Tag</Tag>).toJSON();
  const error = renderer.create(<Tag variation="error">Tag</Tag>).toJSON();
  expect(tag).toMatchSnapshot();
  expect(info).toMatchSnapshot();
  expect(warning).toMatchSnapshot();
  expect(error).toMatchSnapshot();
});
