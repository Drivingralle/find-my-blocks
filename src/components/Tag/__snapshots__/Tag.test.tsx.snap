// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="tag default"
>
  Tag
</div>
`;

exports[`render different variations 1`] = `
<div
  className="tag default"
>
  Tag
</div>
`;

exports[`render different variations 2`] = `
<div
  className="tag info"
>
  Tag
</div>
`;

exports[`render different variations 3`] = `
<div
  className="tag warning"
>
  Tag
</div>
`;

exports[`render different variations 4`] = `
<div
  className="tag error"
>
  Tag
</div>
`;

exports[`render with an icon 1`] = `
<div
  className="tag default hasIcon"
>
  <div
    className="icon"
  >
    <svg
      className="feather feather-play "
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={24}
    >
      <g
        dangerouslySetInnerHTML={
          Object {
            "__html": "<polygon points=\\"5 3 19 12 5 21 5 3\\"></polygon>",
          }
        }
      />
    </svg>
  </div>
  Tag
</div>
`;
